docs/admin/pool_operations.md

```diff
@@ -1,38 +1,124 @@
-DAOS system and is managed by the administrator. The amount of space allocated
-to a pool is decided at creation time and can eventually be expanded through the
-management interface or the `dmg` utility.
+A DAOS pool is a storage reservation that can span any number of storage engines in a
+DAOS system. Pools are managed by the administrator. The amount of space allocated
+to a pool is decided at creation time with the `dmg pool create` command.
+Pools can be expanded at a later time with the `dmg pool expand` command
+that adds additional engine ranks to the existing pool's storage allocation.
+The DAOS management API also provides these capabilities.
+
 
 ## Pool Basics
 
+The `dmg pool` command is the main administrative tool to manage pools.
+Its subcommands can be grouped into the following areas:
+
+* Commands to create, list, query, extend, and destroy a pool.
+  These are the basic functions to manage the storage allocation in a pool.
+
+* Commands to list and set pool properties,
+  and commands to set and list Access Control Lists (ACLs) for a pool.
+
+* Commands to manage failures and other non-standard scenarios.
+  This includes draining, excluding and re-integrating targets,
+  and evicting client connections to a pool.
+
+* An upgrade command to upgrade a pool's format version
+  after a DAOS software upgrade.
+
 ### Creating a Pool
 
-A DAOS pool can be created and destroyed through the `dmg` utility.
+A DAOS pool can be created through the `dmg pool create` command.
+The mandatory parameters that are needed for the creation of a pool
+are the pool label, and a specification of the size of the storage
+allocation.
+
+The pool label must consist of alphanumeric characters, colon (`:`),
+period (`.`), hyphen (`-`) or underscore (`_`).
+The maximum length of a pool label is 127 characters.
+Labels that can be parsed as a UUID (e.g. 123e4567-e89b-12d3-a456-426614174000)
+are forbidden. Pool labels must be unique across the DAOS system.
+
+A pool's size is determined by two factors: How many storage engines are
+participating in the storage allocation, and how much capacity in each
+storage tier is allocated (the latter can be specified either on a per-engine
+basis, or as the total for the pool across all participating engines).
+The same amount of storage will be allocated on each of the participating
+storage engines. If one or more of those engines do not have sufficient free
+space for the requested capacity, the pool creation will fail.
+
+If neither the `--nranks` nor the `--ranks` option is used,
+then the pool will span all storage engines of the DAOS system.
+To limit the pool to only a subset of the engines, those two options
+can be used to specify either the desired number of engines,
+or an explicit list of engine ranks to be used for this pool.
+
+The capacity of the pool can be specified in three different ways:
+
+1. The `--scm-size` parameter (and optionally `--nvme-size`) can
```

Comment:
I would prefer that the `--size` methods be put first, and this method be moved to the bottom. Additionally, I think it would be ideal to note that the `--size` methods are preferred, and that the `--scm-size/--nvme-size` flags should be considered deprecated and are likely to be removed in a future release.

Reply:
I'm willing to be convinced that we shouldn't remove the per-engine method, if there are solid use cases for it. My primary concern is that it doesn't provide a good user experience in the typical use case. A secondary concern is that keeping it means that we also need to continue maintaining and testing it, but those are mostly sunk costs at this point.